File: parnamondal/Week - 2/Day-4/Different Bits Sum Pairwise.py
class Solution:
	# @param A : list of integers
	# @return an integer
	
	# Time Complexity - 0(32N)
	# Space Complexity - O(1)
	
	def cntBits(self, a):
	    #a=[int(bin(i)[2:]) for i in a]
	    s=0
	    m=10**9 + 7
	    for i in range (0,32):
	        c=0
	        for j in range (len(a)):
	            # 'i' left shift by 1
	            if a[j]&(1<<i):
	                c+=1
            s+=(2*c*(len(a)-c))
            s=s%m
        return int(s)

File: parasmittal/DAY 12/Different Bits Sum Pairwise.cpp
int Solution::cntBits(vector<int> &A) {
    long long int count=0;
    for(int i=0;i<32;i++){
        long long int onesum=0,zerosum=0;
        long long int val=1<<i;
        for(int j=0;j<A.size();j++){
            if(A[j] & val)
                onesum++;
            else
                zerosum++;
        }
        count=(count+((2*zerosum*onesum)%1000000007 ))%1000000007;
    }
    return (count)%1000000007;
}
